README.md

```diff
@@ -93,3 +93,135 @@ For simplicity doing calibration and to allow you recalibrate on the fly, you ca
+ln -s ~/.platformio/penv/bin/pio ~/.local/bin/pio
+ln -s ~/.platformio/penv/bin/piodebuggdb ~/.local/bin/piodebuggdb
+```
+
+3. Build and upload the code to the arduino with: `./upload.sh leonardo`
+4. Create a python virtual environment to host mqtt-io:
+
+```
+python -m venv venv
+. ./venv/bin/activate
+pip install mqtt-io
+```
+
+5. Edit the `mqtt-io.yml` in this directory for your system. Ensure you add the MQTT broker and password, and that the stream_modules device matches the device used by `upload.sh` in step 2.
+
+6. Create a systemd file to start monitoring the serial stream, for example:
+
+```
+sudo cat > /etc/systemd/system/fusensors.service <<EOF
+[Unit]
+After=tailscaled.service
+
+[Service]
+WorkingDirectory=/opt/fufarm_arduino_sensors
+ExecStart=/opt/fufarm_arduino_sensors/venv/bin/python3 -m mqtt_io ./mqtt-io.yml
+Restart=always
+StandardOutput=append:/opt/fufarm_arduino_sensors/poll.log
+StandardError=inherit
+SyslogIdentifier=fusensors
+User=fu
+Group=fu
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+sudo systemctl daemon-reload
+sudo systemctl enable fusensors
+sudo systemctl start fusensors
+
+```
+
+7. Create the mqtt.yml file in Home Assitant to expose the sensors from the arduino, e.g:
+
+```
+- sensor:
+    unique_id: farm_temp
+    name: "Farm Temp"
+    state_topic: "sensors/stream/rpiarduino"
+    device_class: "temperature"
+    expire_after: 120
+    value_template: >-
+      {% from 'mqtt_parser.jinja' import parse_stream %}
+      {{ parse_stream(value,'tempair') }}
+
+- sensor:
+    unique_id: farm_humidity
+    name: "Farm Humidity"
+    state_topic: "sensors/stream/rpiarduino"
+    device_class: "humidity"
+    expire_after: 120
+    value_template: >-
+      {% from 'mqtt_parser.jinja' import parse_stream %}
+      {{ parse_stream(value,'humidity') }}
+
+- sensor:
+    unique_id: farm_light
+    name: "Farm Light"
+    state_topic: "sensors/stream/rpiarduino"
+    device_class: "illuminance"
+    expire_after: 120
+    value_template: >-
+      {% from 'mqtt_parser.jinja' import parse_stream %}
+      {{ parse_stream(value,'light') }}
+
+- sensor:
+    unique_id: farm_co2
+    name: "Farm CO2"
+    state_topic: "sensors/stream/rpiarduino"
+    device_class: "carbon_dioxide"
+    expire_after: 120
+    value_template: >-
+      {% from 'mqtt_parser.jinja' import parse_stream %}
+      {{ parse_stream(value,'co2') }}
+
+- sensor:
+    unique_id: farm_flow
+    name: "Sump flow rate"
+    state_topic: "sensors/stream/rpiarduino"
+    device_class: "volume_flow_rate"
+    expire_after: 120
+    value_template: >-
+      {% from 'mqtt_parser.jinja' import parse_stream %}
+      {{ parse_stream(value,'flow') }}
+
+
+- sensor:
+    unique_id: sump_level
+    name: "Sump Level Sensor"
+    state_topic: "sensors/stream/rpiarduino"
+    device_class: "moisture"
+    expire_after: 120
+    value_template: >-
+      {% from 'mqtt_parser.jinja' import parse_stream %}
+      {{ parse_stream(value,'water_level') }}
+```
```

Comment:
Should we put this in `mqtt.yml` in the root of the repository?

Reply:
That's not a bad idea, even if it's only temporary, although I'm working on it currently as I realise that the water level sensors isn't working.

Good shout on the serial output. I've tested it and it works fine currently so I think it's ok for the time being, but I do think moving away from the Arduino Shield is the way forward, it's just it's what I'm currently using on the farm and I've not found a way to support the  DS18B20 sensor directly through mqtt-io yet.

I'll add another commit when I've got the level sensor working